hipcheck/src/plugin/types.rs

```diff
@@ -271,46 +273,103 @@ impl TryFrom<Query> for PluginQuery {
 	}
 }
 
+pub struct MultiplexedQueryReceiver {
+	rx: Streaming<PluginQuery>,
+	backlog: HashMap<i32, VecDeque<PluginQuery>>,
+}
+impl MultiplexedQueryReceiver {
+	pub fn new(rx: Streaming<PluginQuery>) -> Self {
+		Self {
+			rx,
+			backlog: HashMap::new(),
+		}
+	}
+	// @Invariant - this function will never return an empty VecDeque
+	pub async fn recv(&mut self, id: i32) -> Result<Option<VecDeque<PluginQuery>>> {
+		// If we have 1+ messages on backlog for `id`, return them all,
+		// no need to waste time with successive calls
+		if let Some(msgs) = self.backlog.remove(&id) {
+			return Ok(Some(msgs));
+		}
+		// No backlog message, need to operate the receiver
+		loop {
+			let Some(raw) = self.rx.message().await? else {
+				// gRPC channel was closed
+				return Ok(None);
+			};
+			let raw_id = raw.id;
```

Comment:
Why do an assignment here instead of using `raw.id` later?

Reply:
Changed it. It was just because the `insert()` was going to require a `clone()` anyway.